homeassistant/components/weatherflow_cloud/config_flow.py

```diff
@@ -0,0 +1,50 @@
+"""Config flow for WeatherflowCloud integration."""
+from __future__ import annotations
+
+from typing import Any
+
+from aiohttp import ClientResponseError
+import voluptuous as vol
+from weatherflow4py.api import WeatherFlowRestAPI
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_API_TOKEN
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import DOMAIN
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for WeatherFlowCloud."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle a flow initialized by the user."""
+        errors = {}
+        if user_input is not None:
+            self._async_abort_entries_match(user_input)
+
+            # Validate Entry
+            api_token = user_input[CONF_API_TOKEN]
+            try:
+                async with WeatherFlowRestAPI(api_token) as api:
+                    await api.async_get_stations()
+            except ClientResponseError as err:
+                if err.status == 401:
+                    errors["base"] = "invalid_api_key"
```

Comment:
This tells me we should have a `reauth` flow as well and handle that in the coordinator

Reply:
Reauth Added 👏️